rcsb/db/mongo/PdbxLoader.py

```diff
@@ -257,11 +261,16 @@ def __logDocumentSize(self, procName, dList, docIdL):
     def __purgeDocuments(self, databaseName, collectionName, cardinalIdL):
         """Purge documents from collection within database with cardinal identifiers in cardinalIdL."""
         try:
+            # Prepare terminating regex pattern based on database and collection for most efficient searching
+            regexEnd = "$"  # ensures pattern won't overmatch other entries (e.g., bird_chem_comp "PRD" won't match "PRD_000306")
+            if databaseName in ["pdbx_core", "pdbx_comp_model_core"] and "core_entry" not in collectionName:
+                regexEnd = "[_.-]"  # captures entities, instances, and assemblies
+            #
             with Connection(cfgOb=self.__cfgOb, resourceName=self.__resourceName) as client:
                 mg = MongoDbUtil(client)
                 for cardId in cardinalIdL:
                     # selectD = {"rcsb_id": "/%s/" % cardId} # this filter did not work
-                    selectD = {"rcsb_id": {"$regex": "^%s([._-]|$)" % cardId.upper()}}  # case-sensitive (avoid case-insensitive -- very slow performance)
+                    selectD = {"rcsb_id": {"$regex": f"^{cardId.upper()}{regexEnd}"}}  # case-sensitive (avoid case-insensitive -- very slow performance)
```

Comment:
I like this solution, it's a little less messy than using a different separator for each collection. Only issue I can imagine is if in the future a different naming convention containing these characters is used for a new source of CSMs. Just something to keep in mind if this needs to be re-enabled at some point.

Reply:
Thanks! And yes that's true, but I don't think that's something we need to worry about. When incorporating CSMs, a lot of thought and discussion was put into how we format the CSM `rcsb_id`s because of that exact concern. In fact, that's why we are taking the original AF entry IDs (e.g., `AF-A0A1U8FD60-F1`) and modifying it to a new format `AF_AFA0A1U8FD60F1`. This re-labeling all takes place at the point of downloading the CSMs, and the punctuation stripping is applied to every case.

Also, those `[._-]` terminating conventions are so standard across all of our other codebase that if we were to introduce a conflicting case, everything in general would basically explode.

Reply:
Good to know it's already on the radar. Thanks!